test/k8sT/Services.go

```diff
@@ -837,32 +836,20 @@ var _ = Describe("K8sServicesTest", func() {
 		})
 
 		validateEgress := func() {
-			By("Checking that toServices CIDR is plumbed into CEP")
+			By("Checking that toServices CIDR is plumbed into the policy")
 			Eventually(func() string {
-				res := kubectl.Exec(fmt.Sprintf(
-					"%s -n %s get cep %s -o json",
-					helpers.KubectlCmd,
-					helpers.DefaultNamespace,
-					podName))
-				ExpectWithOffset(1, res).Should(helpers.CMDSuccess(), "cannot get Cilium endpoint")
-				data, err := res.Filter(`{.status.policy.egress}`)
-				ExpectWithOffset(1, err).To(BeNil(), "unable to get endpoint %s metadata", podName)
-				return data.String()
+				output, err := kubectl.LoadedPolicyInFirstAgent()
```

Comment:
I think that there's a subtle difference here where previously the endpoint plumbing was tested, but now it's only the policy level plumbing that's tested. Arguably we could achieve the exact same as before using `cilium endpoint list -o json`.

That said, I'm not sure whether this particular detail is necessary to validate in these tests so I'll leave it up to your discretion whether this is worth further modifying.

Reply:
Yes, there is a difference. The rationale here is that we a lot of tests already covering CIDR + label policies to endpoint plumbing combined with connectivity tests. What seemed vital is to test the correct translation of service-based policies to the relevant CIDR policies.